utils/common/src/main/kotlin/EnvironmentVariableFilter.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2022 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils.common
+
+/**
+ * An object providing functionality to filter environments that are passed to newly created processes.
+ *
+ * For many tasks, ORT spawns new processes using the [ProcessCapture] class. When creating a new process, the child
+ * process by default inherits all environment variables from the parent. This could impose a security risk, for
+ * instance if logic in build scripts could theoretically access sensitive information stored in environment
+ * variables, such as database or service credentials.
+ *
+ * To reduce this risk, this object filters the environment variables passed to child processes based on the
+ * following criteria:
+ * - Substrings for variable names can be defined to determine variables with sensitive information. The object provides
+ *   some default strings to match variable names like "PASS", "USER", "TOKEN", etc.
+ * - There is an allow list to include variables even if they contain one of these substrings.
+ *
+ * So in order to determine whether a specific variable "E" can be passed to a child process, this filter applies the
+ * following steps:
+ * - If E is contained in the allow list, it is included.
+ * - Otherwise, E is included if and only if its name does not contain one of the exclusion substrings (ignoring case).
+ */
+object EnvironmentVariableFilter {
```

Comment:
Should these objects not be `internal` ?
Theoretically, I could develop a malicious ORT plugins that modifies these global filters ?

Reply:
Valid point. Currently, the object needs to be public because the filters are initialized by the ORT main function from the configuration. The plan is to change this later with usage of dependency injection.

Reply:
Ok Then maybe add a TODO ?